src/services/fetchers/PolygonIOStockPriceFetcher.ts

```diff
@@ -1,20 +1,21 @@
 import {inject, injectable} from 'inversify';
 
 import PolygonIOStockPriceService from '../PolygonIOStockPriceService.js';
-import {FeedFetcherInterface} from '../../types/fetchers.js';
+import {FeedFetcherInterface, FeedFetcherOptions} from '../../types/fetchers.js';
 
 @injectable()
 class PolygonIOPriceFetcher implements FeedFetcherInterface {
   @inject(PolygonIOStockPriceService) polygonIOStockPriceService!: PolygonIOStockPriceService;
 
-  async apply({sym}: {sym: string}, timestamp: number): Promise<number> {
-    const price = await this.polygonIOStockPriceService.getLatestPrice(sym, timestamp);
+  async apply(params: {sym: string}, options: FeedFetcherOptions): Promise<number> {
+    // TODO: if it is the latest price, does it need a timestamp as argument?
+    const price = await this.polygonIOStockPriceService.getLatestPrice(params.sym, options.timestamp || 0);
```

Comment:
we should never do things like this, where you don't have a data and you manually set some number value like `0`. it was required before here so we can not change logic like that.

Reply:
I agree, this is bad